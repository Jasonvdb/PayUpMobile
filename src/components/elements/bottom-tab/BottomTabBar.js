import React, { Component, Fragment } from "react";
import {
  View,
  StyleSheet,
  Dimensions,
  SafeAreaView,
  ImageBackground
} from "react-native";
import PropTypes from "prop-types";

import CtaButton from "./CtaButton";
import TabButton from "./TabButton";
import theme from "../../../config/theme";

const windowWidth = Dimensions.get("window").width;

const icons = {
  Home: "home",
  Wallet: "wallet",
  Bets: "pie-chart",
  Settings: "settings",
  NewBet: "plus"
};

class BottomTabBar extends Component {
  onTabPressHandler(route) {
    const { onTabPress } = this.props;
    onTabPress({ route });
  }

  onTabLongPressHandler(route) {
    const { onTabLongPress } = this.props;
    onTabLongPress({ route });
  }

  render() {
    const {
      getAccessibilityLabel,
      onTabPress,
      onTabLongPress,
      navigation
    } = this.props;

    const { routes, index: activeRouteIndex } = navigation.state;

    const hideCTA = activeRouteIndex !== 0;

    let rootStyle = styles.root;

    if (hideCTA) {
      rootStyle = { ...rootStyle, ...styles.filledRoot };
    }

    return (
      <View style={rootStyle}>
        <ImageBackground
          style={styles.tabImage}
          source={require("../../../../images/tab.png")}
        >
          <View style={styles.container}>
            {routes.map((route, routeIndex) => {
              const isActive = routeIndex === activeRouteIndex;

              if (routeIndex === 2 && !hideCTA) {
                //TODO move CtaButton into TabButton
                return (
                  <View
                    key={routeIndex}
                    style={{ flex: 1, display: "flex", alignItems: "center" }}
                  >
                    <CtaButton
                      onPress={this.onTabPressHandler.bind(this)}
                      onLongPress={this.onTabLongPressHandler.bind(this)}
                      getAccessibilityLabel={getAccessibilityLabel}
                      route={route}
                      hide={hideCTA}
                    />
                  </View>
                );
              }

              return (
                <TabButton
                  key={routeIndex}
                  onPress={this.onTabPressHandler.bind(this)}
                  onLongPress={this.onTabLongPressHandler.bind(this)}
                  getAccessibilityLabel={getAccessibilityLabel}
                  route={route}
                  iconString={icons[route.key]}
                  isActive={isActive}
                />
              );
            })}
          </View>
        </ImageBackground>
        <SafeAreaView/>
      </View>
    );
  }
}

BottomTabBar.propTypes = {
  onTabPress: PropTypes.func.isRequired,
  onTabLongPress: PropTypes.func.isRequired,
  getAccessibilityLabel: PropTypes.func.isRequired,
  navigation: PropTypes.object.isRequired
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: "transparent",
    shadowColor: "black",
    shadowOffset: { width: 0, height: -14 },
    shadowOpacity: 0.06,
    shadowRadius: 14,
    position: "absolute",
    left: 0,
    right: 0,
    bottom: 0
  },
  filledRoot: {
    backgroundColor: "#FFF"
  },
  container: {
    flexDirection: "row",
    height: theme.bottomTabHeight,
    elevation: 2,
    alignItems: "center"
  },
  tabButton: {
    flex: 1,
    display: "flex",
    justifyContent: "center",
    height: "100%",
    alignItems: "center"
  },
  tabImage: {
    width: windowWidth,
    resizeMode: "center"
  }
});

export default BottomTabBar;
