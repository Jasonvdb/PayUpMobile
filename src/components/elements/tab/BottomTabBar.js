import React, { Component, Fragment } from "react";
import {
	View,
	StyleSheet,
	TouchableOpacity,
	Dimensions,
	SafeAreaView,
	ImageBackground
} from "react-native";
import Icon from "react-native-vector-icons/SimpleLineIcons";
import PropTypes from "prop-types";

import theme from "../../../config/theme";
import CtaButton from "./CtaButton";
import TabButton from "./TabButton";

const windowWidth = Dimensions.get("window").width;

const icons = {
	Home: "home",
	Wallet: "wallet",
	Bets: "pie-chart",
	Settings: "settings"
};

class BottomTabBar extends Component {
	onTabPressHandler(route) {
		const { onTabPress } = this.props;
		onTabPress({ route });
	}

	onTabLongPressHandler(route) {
		const { onTabLongPress } = this.props;
		onTabLongPress({ route });
	}

	render() {
		const {
			getAccessibilityLabel,
			onTabPress,
			onTabLongPress,
			navigation
		} = this.props;

		const { routes, index: activeRouteIndex } = navigation.state;

		return (
			<View style={styles.root}>
				<ImageBackground
					style={styles.tabImage}
					source={require("../../../../images/tab.png")}
				>
					<View style={styles.container}>
						{routes.map((route, routeIndex) => {
							const isActive = routeIndex === activeRouteIndex;
							if (routeIndex === 2) {
								return (
									<CtaButton
										key={routeIndex}
										onPress={this.onTabPressHandler.bind(this)}
										onLongPress={this.onTabLongPressHandler.bind(this)}
										getAccessibilityLabel={getAccessibilityLabel}
										route={route}
										isActive={isActive}
									/>
								);
							}

							return (
								<TabButton
									key={routeIndex}
									onPress={this.onTabPressHandler.bind(this)}
									onLongPress={this.onTabLongPressHandler.bind(this)}
									getAccessibilityLabel={getAccessibilityLabel}
									route={route}
									iconString={icons[route.key]}
									isActive={isActive}
								/>
							);
						})}
					</View>
				</ImageBackground>
				<SafeAreaView/>
			</View>
		);
	}
}

BottomTabBar.propTypes = {
	onTabPress: PropTypes.func.isRequired,
	onTabLongPress: PropTypes.func.isRequired,
	getAccessibilityLabel: PropTypes.func.isRequired,
	navigation: PropTypes.object.isRequired
};

const styles = StyleSheet.create({
	root: {
		backgroundColor: "transparent",
		shadowColor: "black",
		shadowOffset: { width: 0, height: -14 },
		shadowOpacity: 0.06,
		shadowRadius: 14,
		position: "absolute",
		left: 0,
		right: 0,
		bottom: 0
	},
	container: {
		flexDirection: "row",
		height: 60,
		elevation: 2,
		alignItems: "center"
	},
	tabButton: {
		flex: 1,
		display: "flex",
		justifyContent: "center",
		height: "100%",
		alignItems: "center"
	},
	tabImage: {
		width: windowWidth,
		resizeMode: "center"
	}
});

export default BottomTabBar;
