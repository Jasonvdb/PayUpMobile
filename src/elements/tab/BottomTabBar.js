import React, { Fragment } from "react";
import {
	View,
	StyleSheet,
	TouchableOpacity,
	Dimensions,
	SafeAreaView,
	ImageBackground
} from "react-native";
import SimpleLineIcon from "react-native-vector-icons/SimpleLineIcons";
import IonIcon from "react-native-vector-icons/Ionicons";
import theme from "../../config/theme";

const windowWidth = Dimensions.get("window").width;

const styles = StyleSheet.create({
	root: {
		backgroundColor: "transparent",
		shadowColor: "black",
		shadowOffset: { width: 0, height: -14 },
		shadowOpacity: 0.06,
		shadowRadius: 14,
		position: "absolute",
		left: 0,
		right: 0,
		bottom: 0
	},
	container: {
		flexDirection: "row",
		height: 60,
		elevation: 2,
		alignItems: "center"
	},
	tabButton: {
		flex: 1,
		display: "flex",
		justifyContent: "center",
		height: "100%",
		alignItems: "center"
	},
	ctaButton: {
		top: -37,
		backgroundColor: theme.brand1,
		width: 56,
		height: 56,
		borderRadius: 35,
		display: "flex",
		justifyContent: "center",
		alignItems: "center",
		paddingTop: 4,

		shadowOffset: { width: 0, height: 4 },
		shadowOpacity: 0.35,
		shadowRadius: 12,
		shadowColor: theme.brand1
	},
	tabImage: {
		width: windowWidth,
		resizeMode: "center"
	}
});

const icons = {
	Home: "home",
	Wallet: "wallet",
	Bets: "pie-chart",
	Settings: "settings"
};

const CtaButton = () => (
	<TouchableOpacity style={styles.ctaButton}>
		<IonIcon name={"ios-add"} size={35} color={"#FFF"}/>
	</TouchableOpacity>
);

const BottomTabBar = props => {
	const { onTabPress, onTabLongPress, getAccessibilityLabel, navigation } = props;

	const { routes, index: activeRouteIndex } = navigation.state;

	return (
		<View style={styles.root}>
			<ImageBackground
				style={styles.tabImage}
				source={require("../../../images/tab.png")}
			>
				<View style={styles.container}>
					{routes.map((route, routeIndex) => {
						const isRouteActive = routeIndex === activeRouteIndex;

						return (
							<Fragment key={routeIndex}>
								{routeIndex === 2 ? <CtaButton/> : null}
								<TouchableOpacity
									style={styles.tabButton}
									onPress={() => {
										onTabPress({ route });
									}}
									onLongPress={() => {
										onTabLongPress({ route });
									}}
									accessibilityLabel={getAccessibilityLabel({
										route
									})}
								>
									<SimpleLineIcon
										name={icons[route.key]}
										size={25}
										color={isRouteActive ? theme.brand1 : theme.disabled1}
									/>
								</TouchableOpacity>
							</Fragment>
						);
					})}
				</View>
			</ImageBackground>
			<SafeAreaView/>
		</View>
	);
};

export default BottomTabBar;
