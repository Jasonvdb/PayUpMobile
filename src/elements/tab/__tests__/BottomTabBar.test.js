/**
 * @format
 */

import "react-native";
import React from "react";
import { shallow, render } from "enzyme";

import BottomTabBar from "../BottomTabBar";

import renderer from "react-test-renderer";
import { Text, View } from "react-native";

const DummyScreen = () => (
	<View>
		<Text>Dummy screen</Text>
	</View>
);

const testProps = {
	onTabPress: () => {},
	onTabLongPress: () => {},
	getAccessibilityLabel: () => {},
	navigation: {
		state: {
			routes: [
				{
					key: "HomeTest",
					isTransitioning: false,
					index: 0,
					routes: [{ routeName: "Home", key: "id-1566063909546-0" }],
					routeName: "Home"
				},
				{
					key: "WalletTest",
					isTransitioning: false,
					index: 0,
					routes: [{ routeName: "Wallet", key: "id-1566063909546-1" }],
					routeName: "Wallet"
				},
				{
					key: "BetsTest",
					isTransitioning: false,
					index: 0,
					routes: [{ routeName: "Bets", key: "id-1566063909546-2" }],
					routeName: "Bets"
				},
				{
					key: "SettingsTest",
					isTransitioning: false,
					index: 0,
					routes: [{ routeName: "Settings", key: "id-1566063909546-3" }],
					routeName: "Settings"
				}
			],
			index: 0
		}
	}
};

describe("BottomTabBar", () => {
	describe("Rendering", () => {
		it("should render", () => {
			//renderer.create(<BottomTabBar {...testProps}/>);
		});

		it("should match to snapshot", () => {
			const component = shallow(<BottomTabBar {...testProps}/>);
			expect(component).toMatchSnapshot();
		});
	});
});
