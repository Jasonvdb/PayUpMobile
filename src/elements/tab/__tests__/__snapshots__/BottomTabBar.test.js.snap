// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabBar Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
